runtime/src/commonMain/kotlin/pbandk/internal/json/JsonMessageMarshaller.kt

```diff
@@ -38,6 +38,7 @@ internal class JsonMessageMarshaller(private val jsonConfig: JsonConfig) : Messa
             is BytesValue -> writeWrapperValue(message, message.value)
             // Other well-known types with special JSON encoding
             is Timestamp -> jsonValueMarshaller.writeString(Util.timestampToString(message))
+            is Struct -> jsonValueMarshaller.writeStruct(message)
```

Comment:
🔧 I think you can replace your custom `writeStruct()` method with `jsonValueMarshaller.writeValue(message.fields, message.fieldDescriptors[0].type)`. Basically, the JSON representation of `Struct` is the representation of the `Struct.fields` map.

Then instead of the `Value.toElement()` extension method you should add `Value` as another case to this `when` statement. That will make sure that this code can properly serialize `Value` objects even if they're used outside of ` `Struct`. You'll know you did it right when the conformance tests that relate to `Value` are passing.

Likewise, you should do the same in the unmarshalling code: use the existing `readMap()` method to populate the `Struct.fields` field. And add handling of `Value` alongside the handling of `Struct` in the `when` statement in `JsonValueUnmarshaller`.